### PR TITLE
feat(remix-dev): allow importing `.csv` files

### PR DESCRIPTION
**Pull request by kayac-chang**
Discussion: https://github.com/remix-run/remix/discussions/3919

- [ ] Docs
- [ ] Tests

Testing Strategy:

### CODE REVIEW

**Comment by MichaelDeBoey** [comment]
@kayac-chang Could you please keep the map alphabetically?

**Comment by kayac-chang** [comment]
> @kayac-chang Could you please keep the map alphabetically?

roger

File: .changeset/wise-masks-sniff.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"remix": patch
```
Comment (MichaelDeBoey) [inline .changeset/wise-masks-sniff.md:2]:
```suggestion
```

File: .changeset/wise-masks-sniff.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+"@remix-run/dev": patch
```
Comment (MichaelDeBoey) [inline .changeset/wise-masks-sniff.md:3]: This is a feature, not a bugfix

```suggestion
"@remix-run/dev": minor
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

File: .changeset/wise-masks-sniff.md (hunk 2 above)
Comment (kayac-chang) [reply .changeset/wise-masks-sniff.md:3]: sorry, fix now 😱

**Comment by kayac-chang** [comment]
Hello, @MichaelDeBoey 
I run `yarn test` on local environment, 
I got three testing failed 
```
`replace-remix-imports` migration › runs successfully
remix CLI › create prompts › allows you to go through the prompts and convert to JS
`convert-to-javascript` migration › runs successfully when ran programmatically
```
but I have no idea how to fixed that

**Comment by MichaelDeBoey** [comment]
@kayac-chang All tests are passing both locally (macOS) & on CI
Only Windows has some "failing" tests, but this is due to timeouts

**Comment by kayac-chang** [comment]
> @kayac-chang All tests are passing both locally (macOS) & on CI Only Windows has some "failing" tests, but this is due to timeouts

Yeah🥳 
Glad to heard that!!!
Is there anything I can help?

**Comment by MichaelDeBoey** [comment]
@kayac-chang @mcansh is looking into the Windows failures in #3915, so we'll get there 👍

File: packages/remix-dev/compiler/loaders.ts [hunk 3]
```diff
@@ -5,6 +5,7 @@ export const loaders: { [ext: string]: esbuild.Loader } = {
   ".aac": "file",
   ".avif": "file",
   ".css": "file",
+  ".csv": "file",
```
Comment (MichaelDeBoey) [inline packages/remix-dev/compiler/loaders.ts:8]: We should have an entry in `packages/remix-dev/modules.ts` as well

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

File: packages/remix-dev/compiler/loaders.ts (hunk 3 above)
Comment (kayac-chang) [reply packages/remix-dev/compiler/loaders.ts:8]: @MichaelDeBoey sorry for late reply, I'm already add `.csv` into `packages/remix-dev/modules.ts`

**Comment by MichaelDeBoey** [comment]
@kayac-chang Could you please look at the discussions I started?

**Comment by kayac-chang** [comment]
> needs-response

Sorry @MichaelDeBoey 
Which discussion I need to response?

**Comment by kayac-chang** [comment]
@MichaelDeBoey 
sorry I missing your message.
I change the `wise-masks-sniff.md` to `"@remix-run/dev": minor` now.

**Review by machour — APPROVED** [review]

File: .changeset/wise-masks-sniff.md (hunk 1 above)
Comment (chaance) [reply .changeset/wise-masks-sniff.md:2]: @MichaelDeBoey I'd prefer we not use these sort of changes as a basis for a minor release. I see this as any file that doesn't require its own special loader should be loaded as a file, so any files that we don't currently map to that should be considered a bug. Of course adding loader rules in perpetuity isn't a good long-term solution, but that's a separate conversation.

File: .changeset/wise-masks-sniff.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+"remix": patch
+"@remix-run/dev": minor
```
Comment (chaance) [inline .changeset/wise-masks-sniff.md:3]:
```suggestion
"@remix-run/dev": patch
```

File: .changeset/wise-masks-sniff.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+---
+"remix": patch
+"@remix-run/dev": patch
+---
+
+feat(remix-dev/compiler): import `.csv` files as file
```
Comment (chaance) [inline .changeset/wise-masks-sniff.md:6]:
```suggestion
Add file loader for importing `.csv` files
```

**Review by chaance — APPROVED** [review]

**Pull request merged**